### PR TITLE
fix(hstore): JRaft Histogram Metrics Value NaN

### LINKED ISSUE #2627: [Bug] HStore JRaft Histograms Metrics Value NaN Error
### Bug Type (问题类型)

None

### Before submit

- [X] 我已经确认现有的 [Issues](https://github.com/apache/hugegraph/issues) 与 [FAQ](https://hugegraph.apache.org/docs/guides/faq/) 中没有相同 / 重复问题 (I have confirmed and searched that there are no similar problems in the historical issue and documents)

### Environment (环境信息)

- Server Version: Master (Apache Release Version)
- Backend: HStore

### Expected & Actual behavior (期望与实际表现)

# 当前问题
Histograms 类型指标值获取出现异常 NaN
```
jraft_append_logs_bytes_summary_count{group="0",handle="data",hg="store",} NaN
# HELP jraft_append_logs_bytes_mean  
# TYPE jraft_append_logs_bytes_mean gauge
jraft_append_logs_bytes_mean{group="0",handle="data",hg="store",} NaN
# HELP jraft_append_logs_bytes_min  
# TYPE jraft_append_logs_bytes_min gauge
jraft_append_logs_bytes_min{group="0",handle="data",hg="store",} NaN
# HELP jraft_append_logs_bytes_median  
# TYPE jraft_append_logs_bytes_median gauge
jraft_append_logs_bytes_median{group="0",handle="data",hg="store",} NaN
# HELP jraft_append_logs_bytes_max  
# TYPE jraft_append_logs_bytes_max gauge
jraft_append_logs_bytes_max{group="0",handle="data",hg="store",} NaN
# HELP jraft_append_logs_bytes_summary_sum  
# TYPE jraft_append_logs_bytes_summary_sum gauge
jraft_append_logs_bytes_summary_sum{group="0",handle="data",hg="store",} NaN
# HELP jraft_append_logs_bytes_summary  
# TYPE jraft_append_logs_bytes_summary gauge
jraft_append_logs_bytes_summary{group="0",handle="data",hg="store",quantile="0.95",} NaN
jraft_append_logs_bytes_summary{group="0",handle="data",hg="store",quantile="0.75",} NaN
jraft_append_logs_bytes_summary{group="0",handle="data",hg="store",quantile="0.999",} NaN
jraft_append_logs_bytes_summary{group="0",handle="data",hg="store",quantile="0.98",} NaN
jraft_append_logs_bytes_summary{group="0",handle="data",hg="store",quantile="0.99",} NaN
jraft_append_logs_bytes_summary{group="0",handle="data",hg="store",quantile="0.5",} NaN

```

# 期望结果
通过Spring Actuator 接口获取正确的JRaft 监控指标

curl http://ip:8620/actuator/prometheus |grep jraft_append_logs_bytes

# 指标详情
```
-- <hg_0/127.0.0.1:8510> -- Histograms ------------------------------------------------------------------
append-logs-bytes
             count = 67710
               min = 110
               max = 110
              mean = 110.00
            stddev = 0.00
            median = 110.00
              75% <= 110.00
              95% <= 110.00
              98% <= 110.00
              99% <= 110.00
            99.9% <= 110.00
append-logs-count
             count = 67710
               min = 1
               max = 1
              mean = 1.00
            stddev = 0.00
            median = 1.00
              75% <= 1.00
              95% <= 1.00
              98% <= 1.00
              99% <= 1.00
            99.9% <= 1.00
fsm-apply-tasks-count
             count = 67705
               min = 1
               max = 1
              mean = 1.00
            stddev = 0.00
            median = 1.00
              75% <= 1.00
              95% <= 1.00
              98% <= 1.00
              99% <= 1.00
            99.9% <= 1.00
handle-append-entries-count
             count = 20
               min = 0
               max = 0
              mean = 0.00
            stddev = 0.00
            median = 0.00
              75% <= 0.00
              95% <= 0.00
              98% <= 0.00
              99% <= 0.00
            99.9% <= 0.00
replicate-entries-bytes
             count = 135378
               min = 110
               max = 110
              mean = 110.00
            stddev = 0.00
            median = 110.00
              75% <= 110.00
              95% <= 110.00
              98% <= 110.00
              99% <= 110.00
            99.9% <= 110.00
replicate-entries-count
             count = 135378
               min = 1
               max = 1
              mean = 1.00
            stddev = 0.00
            median = 1.00
              75% <= 1.00
              95% <= 1.00
              98% <= 1.00
              99% <= 1.00
            99.9% <= 1.00
replicator-hg_0/127.0.0.1:8512::64.replicate-inflights-count
             count = 20365
               min = 1
               max = 1
              mean = 1.00
            stddev = 0.00
            median = 1.00
              75% <= 1.00
              95% <= 1.00
              98% <= 1.00
              99% <= 1.00
            99.9% <= 1.00
replicator-hg_0/127.0.0.1:8513::80.replicate-inflights-count
             count = 20365
               min = 1
               max = 1
              mean = 1.00
            stddev = 0.00
            median = 1.00
              75% <= 1.00
              95% <= 1.00
              98% <= 1.00
              99% <= 1.00
            99.9% <= 1.00

```

# 具体指标
```
append-logs-bytes
append-logs-count
fsm-apply-tasks-count
handle-append-entries-count
replicate-entries-bytes
replicate-entries-count
replicate-inflights-count
```

### Vertex/Edge example (问题点 / 边数据举例)

_No response_

### Schema [VertexLabel, EdgeLabel, IndexLabel] (元数据结构)

_No response_

### PR DESCRIPTION
**Pull request by JackyYangPassion**
## Purpose of the PR

- close #2627  

## Verifying these changes

- [ ] Trivial rework / code cleanup without any test coverage. (No Need)
- [ ] Already covered by existing tests, such as *(please modify tests here)*.
- [ ] Need tests and can be verified as follows:
    - xxx

## Does this PR potentially affect the following parts?

- [ ]  Dependencies ([add/update license](https://hugegraph.apache.org/docs/contribution-guidelines/contribute/#321-check-licenses) info & [regenerate_dependencies](../hugegraph-server/hugegraph-dist/scripts/dependency/regenerate_known_dependencies.sh)) 
- [ ]  Modify configurations
- [ ]  The public API
- [ ]  Other affects (typed here)
- [ ]  Nope

## Documentation Status

- [ ]  `Doc - TODO` 
- [ ]  `Doc - Done` 
- [ ]  `Doc - No Need`

### CODE REVIEW

File: hugegraph-store/hg-store-node/src/main/java/org/apache/hugegraph/store/node/metrics/JRaftMetrics.java [hunk 1]
```diff
@@ -172,35 +168,33 @@ private static void registerHistogram(String group, String name,
 
         String baseName = PREFIX + "." + name.toLowerCase();
 
-        HistogramWrapper wrapper = toWrapper(histogram);
-
-        Gauge.builder(baseName + ".median", wrapper, (d) -> d.getSnapshot().getMedian())
+        Gauge.builder(baseName + ".median", histogram, (d) -> d.getSnapshot().getMedian())
```
Comment (javeme) [inline hugegraph-store/hg-store-node/src/main/java/org/apache/hugegraph/store/node/metrics/JRaftMetrics.java:171]: Will getSnapshot() be called frequently?

Comment (JackyYangPassion) [reply hugegraph-store/hg-store-node/src/main/java/org/apache/hugegraph/store/node/metrics/JRaftMetrics.java:171]: yes, request pull metrics from Prometheus every time

Comment (JackyYangPassion) [reply hugegraph-store/hg-store-node/src/main/java/org/apache/hugegraph/store/node/metrics/JRaftMetrics.java:171]: It turns out that the logic is to only request a snapshot after > 30 seconds, 
which will cause the indicator to be empty. Therefore,
 it is requested once every time the indicator is pulled. 
The frequency is not high. 

Online, it is usually pulled once every 15 seconds.

Comment (javeme) [reply hugegraph-store/hg-store-node/src/main/java/org/apache/hugegraph/store/node/metrics/JRaftMetrics.java:171]: ok

File: hugegraph-store/hg-store-node/src/main/java/org/apache/hugegraph/store/node/metrics/JRaftMetrics.java [hunk 2]
```diff
@@ -172,35 +168,33 @@ private static void registerHistogram(String group, String name,
...
              .tags(tags).register(registry);
-        Gauge.builder(baseName + ".min", wrapper, (d) -> d.getSnapshot().getMin())
+        Gauge.builder(baseName + ".min", histogram, (d) -> d.getSnapshot().getMin())
```
Comment (javeme) [inline hugegraph-store/hg-store-node/src/main/java/org/apache/hugegraph/store/node/metrics/JRaftMetrics.java:173]: we can also rename "(d) ->" to "h ->"

File: hugegraph-store/hg-store-node/src/main/java/org/apache/hugegraph/store/node/metrics/JRaftMetrics.java [hunk 3]
```diff
@@ -172,35 +168,33 @@ private static void registerHistogram(String group, String name,
...
              .tags(tags).register(registry);
-        Gauge.builder(baseName + ".max", wrapper, (d) -> d.getSnapshot().getMax())
+        Gauge.builder(baseName + ".max", histogram, (d) -> d.getSnapshot().getMax())
```
Comment (javeme) [inline hugegraph-store/hg-store-node/src/main/java/org/apache/hugegraph/store/node/metrics/JRaftMetrics.java:175]: A further question, if each dimension like `min/max/mean` needs to be retrieved getSnapshot() once, does it need to be optimized?

File: hugegraph-store/hg-store-node/src/main/java/org/apache/hugegraph/store/node/metrics/JRaftMetrics.java (hunk 2 above)
Comment (JackyYangPassion) [reply hugegraph-store/hg-store-node/src/main/java/org/apache/hugegraph/store/node/metrics/JRaftMetrics.java:173]: Thanks for your suggestion, I will optimize it

File: hugegraph-store/hg-store-node/src/main/java/org/apache/hugegraph/store/node/metrics/JRaftMetrics.java (hunk 3 above)
Comment (JackyYangPassion) [reply hugegraph-store/hg-store-node/src/main/java/org/apache/hugegraph/store/node/metrics/JRaftMetrics.java]: No, each time you pull indicators, you need to put the latest one into the indicator system.

**Review by imbajin — APPROVED** [review]

**Review by VGalaxies — APPROVED** [review]

**Pull request merged**